ext/pdo/pdo_stmt.c

```diff
@@ -1389,6 +1394,9 @@ PHP_METHOD(PDOStatement, fetchAll)
 	}
 
 	do_fetch_opt_finish(stmt, 0);
+	if (increase_refcount_ctor) {
```

Comment:
I suppose it's theoretically possible that the ctor_args is changed by user code during the do_fetch operation.
In that case we may be destroying the wrong instance.
So I think you should keep a pointer to the HashTable that you initially refcount-incremented.

Reply:
Actually this seems like a more sensible approach, and more general let me try some new test cases and this idea.